parquet_file/src/utils.rs

```diff
@@ -436,13 +486,48 @@ fn make_record_batch() -> (Vec<RecordBatch>, Schema, Vec<ColumnSummary>, Timesta
         let record_batch =
             RecordBatch::try_new(Arc::clone(schema.inner()), record_batch.columns().to_vec())
                 .expect("record-batch re-creation");
+        num_rows += record_batch.num_rows();
         record_batches.push(record_batch);
     }
 
-    (record_batches, schema, summaries, timestamp_range)
+    (record_batches, schema, summaries, timestamp_range, num_rows)
 }
 
 /// Creates new in-memory object store for testing.
 pub fn make_object_store() -> Arc<ObjectStore> {
     Arc::new(ObjectStore::new_in_memory(InMemory::new()))
 }
+
+pub fn read_data_from_parquet_data(schema: SchemaRef, parquet_data: Vec<u8>) -> Vec<RecordBatch> {
+    let mut record_batches = vec![];
+
+    let cursor = SliceableCursor::new(parquet_data);
+    let reader = SerializedFileReader::new(cursor).unwrap();
+    let mut arrow_reader = ParquetFileArrowReader::new(Arc::new(reader));
+
+    // Indices of columns in the schema needed to read
+    let projection: Vec<usize> = Storage::column_indices(Selection::All, Arc::clone(&schema));
```

Comment:
This type annotation isn't necessary either, if you like having it here, that's fine though :)
```suggestion
    let projection = Storage::column_indices(Selection::All, Arc::clone(&schema));
```